### PR TITLE
Charlesmchen/send fake contacts

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
This PR is now ready for review. 

PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/DebugUI/DebugUIMessages.m [hunk 1]
```diff
@@ -124,6 +124,10 @@ - (nullable OWSTableSection *)sectionForThread:(nullable TSThread *)thread
                         actionBlock:^{
                             [DebugUIMessages selectBackDatedAction:thread];
                         }],
+        [OWSTableItem itemWithTitle:@"Send All Contact Shares"
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/DebugUI/DebugUIMessages.m]: This command actually sends the contact shares.

File: SignalServiceKit/src/Messages/Interactions/OWSContact.h [hunk 2]
```diff
@@ -112,6 +124,8 @@ typedef NS_ENUM(NSUInteger, OWSContactAddressType) {
 
 - (BOOL)ows_isValid;
 
+- (NSString *)debugDescription;
```
Comment (charlesmchen-signal) [inline SignalServiceKit/src/Messages/Interactions/OWSContact.h]: Added `debugDescription`.

File: SignalServiceKit/src/Messages/OWSMessageManager.m [hunk 3]
```diff
@@ -1028,7 +1027,7 @@ - (TSIncomingMessage *_Nullable)handleReceivedEnvelope:(OWSSignalServiceProtosEn
             }
         }
     } else {
-        if (body.length == 0 && attachmentIds.count < 1) {
+        if (body.length == 0 && attachmentIds.count < 1 && !contact) {
```
Comment (michaelkirk-signal) [inline SignalServiceKit/src/Messages/OWSMessageManager.m]: Good catch!

We need something similar in OWSRecordTranscriptJob.m

```
    if (outgoingMessage.body.length < 1 && outgoingMessage.attachmentIds.count < 1) {
        OWSFail(@"Ignoring message transcript for empty message.");
        return;
    }
```

Comment (charlesmchen-signal) [reply SignalServiceKit/src/Messages/OWSMessageManager.m]: Done.

**Pull request merged**